.devcontainer/devcontainer.json

```diff
@@ -0,0 +1,33 @@
+// For format details, see https://aka.ms/devcontainer.json. For config options, see the README at:
+// https://github.com/microsoft/vscode-dev-containers/tree/v0.187.0/containers/rust
+{
+	"name": "Rust",
+    "dockerComposeFile": ["../docker-compose.yaml"],
+    "service": "rust-client",
+	"workspaceFolder": "/workspace/rust-socketio",
+    "shutdownAction": "stopCompose",
+
+	// Set *default* container specific settings.json values on container create.
+	"settings": { 
+		"lldb.executable": "/usr/bin/lldb",
+		// VS Code don't watch files under ./target
+		"files.watcherExclude": {
+			"**/target/**": true
+		}
+	},
+
+	// Add the IDs of extensions you want installed when the container is created.
+	"extensions": [
+		"rust-lang.rust",
```

Comment:
I usually work with rust-analyzer as it generally worked better for me. Could you maybe evaluate why you think the rust plugin is more appropriate here?

Reply:
I don't have a strong preference. I'll switch to rust-analyzer. One small thing that rust-lang.rust does that rust-anaylzer doesn't is the directly running of individual tests.

Reply:
rust-anaylzer is missing basic tab completion functionality (auto-complete use, method names, etc). Perhaps rust-lang.rust has gotten better since you last tried it?

Reply:
Nah that't can't be right, rust-analyzer actually has great auto-completion, even over match clauses. Maybe their language server wasn't set up correctly inside the devcontainer. But anyways that's fine for now, I would change it as soon as I work with it and miss something ;)